### PR TITLE
Remove the choiceObservers feature flag in the engine

### PR DESCRIPTION
**Pull request by paulbrauner-da**
Context: https://github.com/digital-asset/daml/issues/18240

### CODE REVIEW

File: daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV2.scala [hunk 1]
```diff
@@ -707,13 +707,11 @@ private[archive] class DecodeV2(minor: LV.Minor) {
...
             if (lfChoice.hasObservers) {
-              assertSince(LV.Features.choiceObservers, "TemplateChoice.observers")
               decodeExpr(lfChoice.getObservers, s"$tpl:$chName:observers") { observers =>
                 Ret(Some(observers))
               }
             } else {
-              assertUntil(LV.Features.choiceObservers, "missing TemplateChoice.observers")
-              Ret(None)
+              throw notSupportedError("missing TemplateChoice.observers")
             }
```
Comment (remyhaemmerle-da) [inline daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV2.scala:715]:
```suggestion        
              decodeExpr(lfChoice.getObservers, s"$tpl:$chName:observers") { observers =>
                Ret(Some(observers))
              }
```
We should be consistent with the rest. We do not check the required field are defined (e.g. line 707 we have a `lfChoice.getControllers`, without check that `lfChoice.hasControllers`) so we we should do the same here. 

I am not against we systematically verify that *all*  the required fields are actually defined, but this should be done in another PR.

**Review by remyhaemmerle-da — APPROVED** [review]

File: daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV2.scala (hunk 1 above)
Comment (paulbrauner-da) [reply daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV2.scala]: Good point, applied.
